cpp/src/arrow/compute/exec_test.cc

```diff
@@ -55,6 +55,19 @@ using ::arrow::internal::BitmapEquals;
 using ::arrow::internal::CopyBitmap;
 using ::arrow::internal::CountSetBits;
 
+TEST(ExecBatch, Basics) {
+  auto i32_array = ArrayFromJSON(int32(), "[0, 1, 2]");
+  auto utf8_array = ArrayFromJSON(utf8(), R"(["a", "b", "c"])");
+  ExecBatch exec_batch({Datum(i32_array), Datum(utf8_array)}, 3);
+  auto right_schema = schema({field("a", int32()), field("b", utf8())});
+  ASSERT_OK_AND_ASSIGN(auto right_record_batch, exec_batch.ToRecordBatch(right_schema));
+  auto accept_schema = schema({field("a", int32())});
+  ASSERT_OK_AND_ASSIGN(auto accept_record_batch, exec_batch.ToRecordBatch(accept_schema));
```

Comment:
Why would this succeed and produce a truncated result here?

Reply:
[This comment](https://github.com/apache/arrow/pull/14386#issuecomment-1276240482) gives the background.

Reply:
Yes, I checked this locally and it occurs in multiple places unfortunately. However, I don't think this is a behavior that we want to set in stone, so I think we should remove this particular test.

Reply:
I'll soon add a commit with checks showing the current situation, if only for posterity, and then I'll remove ones we do not want to keep.